python/ml4ir/base/model/architectures/dnn.py

```diff
@@ -61,18 +77,35 @@ def get_op(layer_type, layer_args):
 
     def get_architecture_op(self):
         def _architecture_op(ranking_features):
-            layer_input = ranking_features
+            if 'positional_bias_handler' in self.model_config and self.model_config['positional_bias_handler']['key']==PositionalBiasHandler.MLRANKER:
+                layer_input = ranking_features[0]
+                # Pass ranking features through all the layers of the DNN
+                for layer_op in self.layer_ops:
+                    layer_input = layer_op(layer_input)
 
-            # Pass ranking features through all the layers of the DNN
-            for layer_op in self.layer_ops:
-                layer_input = layer_op(layer_input)
+                #positional_bias = self.positional_bias_layer(ranking_features[1])
+                positional_bias = self.positional_bias_layer(tf.transpose(ranking_features[1], [0, 2, 1]))
+                scores = layers.Add()([positional_bias, layer_input])
+
+                # Collapse extra dimensions
+                if isinstance(self.layer_ops[-1], layers.Dense) and (self.layer_ops[-1].units == 1):
+                    scores = tf.squeeze(scores, axis=-1)
+
+                return scores
 
-            # Collapse extra dimensions
-            if isinstance(self.layer_ops[-1], layers.Dense) and (self.layer_ops[-1].units == 1):
-                scores = tf.squeeze(layer_input, axis=-1)
             else:
-                scores = layer_input
+                layer_input = ranking_features
+
+                # Pass ranking features through all the layers of the DNN
+                for layer_op in self.layer_ops:
+                    layer_input = layer_op(layer_input)
+
+                # Collapse extra dimensions
+                if isinstance(self.layer_ops[-1], layers.Dense) and (self.layer_ops[-1].units == 1):
+                    scores = tf.squeeze(layer_input, axis=-1)
+                else:
+                    scores = layer_input
 
-            return scores
+                return scores
```

Comment:
@mohazahran You should move the actual implementation of the positional bias to a module under `applications/ranking/model/positional_bias`.